src/modules/job-manager/purge.c

```diff
@@ -0,0 +1,381 @@
+                             "{s:f s:i s:i}",
+                             "age_limit", &age_limit,
+                             "num_limit", &num_limit,
+                             "batch", &batch) < 0)
+        goto error;
+    if (age_limit != INACTIVE_AGE_UNLIMITED && age_limit < 0) {
+        errmsg = "age limit must be >= 0 (0=unlimited)";
+        errno = EINVAL;
+        goto error;
+    }
+    if (num_limit != INACTIVE_NUM_UNLIMITED && num_limit < 0) {
+        errmsg = "num limit must be >= 0 (0=unlimited)";
+        errno = EINVAL;
+        goto error;
+    }
+    if (batch < 1 || batch > purge_batch_max) {
+        errprintf (&error, "batch must be >= 1 and <= %d", purge_batch_max);
+        errmsg = error.text;
+        errno = EINVAL;
+        goto error;
+    }
+    if (!(f = purge_inactive_jobs (purge,
+                                   age_limit,
+                                   num_limit,
+                                   batch))
+        || flux_future_then (f, -1, purge_request_continuation, purge) < 0) {
+        flux_future_destroy (f);
+        if (errno == ENODATA) { // ENODATA means zero jobs can be purged
+            if (flux_respond_pack (h, msg, "{s:i}", "count", 0) < 0)
+                flux_log_error (h, "error responding to purge request");
+            return;
+        }
+        goto error;
+    }
+    if (flux_msg_aux_set (msg,
+                          "future",
+                          f,
+                          (flux_free_f)flux_future_destroy) < 0) {
+        flux_future_destroy (f);
+        goto error;
+    }
+    if (flux_msglist_append (purge->requests, msg) < 0)
+        goto error; // future destroyed with msg by dispatcher
+    return;
+error:
+    if (flux_respond_error (h, msg, errno, errmsg) < 0)
+        flux_log_error (h, "error responding to purge request");
+}
+
+static int purge_parse_config (const flux_conf_t *conf,
+                               flux_error_t *error,
+                               void *arg)
+{
+    struct purge *purge = arg;
+    flux_error_t e;
+    const char *fsd = NULL;
+    double age_limit = INACTIVE_AGE_UNLIMITED;
+    int num_limit = INACTIVE_NUM_UNLIMITED;
+
+    if (flux_conf_unpack (conf,
+                          &e,
+                          "{s?{s?s s?i}}",
+                          "job-manager",
+                            "inactive-age-limit", &fsd,
+                            "inactive-num-limit", &num_limit) < 0)
+        return errprintf (error, "job-manager.max-inactive-*: %s", e.text);
+    if (fsd) {
+        double t;
+        if (fsd_parse_duration (fsd, &t) < 0)
+            return errprintf (error,
+                              "job-manager.inactive-age-limit: invalid FSD");
+        age_limit = t;
+    }
+    if (num_limit != INACTIVE_NUM_UNLIMITED) {
+        if (num_limit < 0)
+            return errprintf (error,
+                              "job-manager.inactive-num-limit: must be >= 0");
```

Comment:
Should the error message be `job-manager.inactive-num-limit: must be >= -1`?

Reply:
Actually, for the TOML config my thought was that users wouldn't specify a limit if they don't want one.  Does that sound OK?

O/w we should probably define a special value for unlimited age too, like "unlimited".

Reply:
That makes sense to me.